compiler/src/iree/compiler/Codegen/Common/GPU/GPULowerToUKernels.cpp

```diff
@@ -131,6 +132,39 @@ static Value createSharedMemory(PatternRewriter &rewriter, Location loc,
   return allocOp;
 }
 
+/// Returns the index of the innermost CrossIntrinsic dimension of the C matrix,
+/// if it is static, and std::nullopt if it is dynamic or if there are no
+/// CrossIntrinsic dims.
+static std::optional<unsigned>
+getCInnermostStaticCrossIntrinsicDim(IREE::Codegen::InnerTiledOp op) {
+  auto outputType = dyn_cast<ShapedType>(op.getResultTypes()[0]);
+  if (!outputType) {
+    return std::nullopt;
+  }
+  auto mma = cast<IREE::GPU::DataTiledMMAAttr>(op.getKind());
+  IREE::Codegen::TileSwizzle accSwizzle =
+      getSwizzle(mma, IREE::GPU::MMAFragment::Acc);
+  SmallVector<IREE::Codegen::TileSwizzle::Dim> swizzleDims;
+  for (IREE::Codegen::TileSwizzle::ExpandShapeDimVectorType group :
+       accSwizzle.expandShape) {
+    swizzleDims.append(group);
+  }
+  applyPermutationToVector(swizzleDims, accSwizzle.permutation);
+  int rankDiff = outputType.getRank() - swizzleDims.size();
+  auto crossIntrinsic = IREE::Codegen::TileSwizzle::Dim::Kind::CrossIntrinsic;
+  for (int swizzleIdx = swizzleDims.size() - 1; swizzleIdx >= 0; --swizzleIdx) {
```

Comment:
FYI, I think the canonical way to index in reverse is `for (size_t e = swizzle.Dims.size(), i = e - 1; i < e; --i)`. This doesn't require you to go to signed integers.